modules/javafx.web/src/main/java/com/sun/webkit/graphics/WCRenderQueue.java

```diff
@@ -171,6 +171,15 @@ public boolean isOpaque() {
         return opaque;
     }
 
+    /*is called from native*/
+    private boolean strokeContains(WCPath path, float x, float y) {
```

Comment:
Though it works, It is weird to have a totally unrelated method in RenderQueue. Rather I would consider to expose a GraphicsContext from RenderQueue and add some helper class(say WCPathHelper) to do the same.

@kevinrushforth, What is your opinion on this?

(Unrelated: This method must be synchronized like other methods of WCRenderQueue)

Reply:
The problem is, that the RenderQueue needs to be flushed/decoded before the GraphicsContext can be queried for the containment test. In an ideal world, `SVGRenderSupport::applyStrokeStyleToContext` would work against a minimal interface, that is just implemented by `GraphicsContext` and we could just supply a minimal implementation, that buffers the set values. I tried another approach:

https://github.com/matthiasblaesing/openjdk-jfx/commit/45a7c0a7f21882054578b802ab16b7ede555fd9c

In that implementation the `GraphicsContext` is modified, to buffer the set stroke properties in the `PlatformContextJava`. The `PlatformContextJava` information is then used to create the `BasicStroke` and do the `strokeContains` check.

Reply:
This seems like the biggest unresolved question: should we go with an approach that requires flushing the render queue and querying the context for the containment test, or go with the other approach to shadow the needed properties in the GraphicsContext. In the latter case, there is no need to touch the rendering. My gut feel is that the latter approach might be better, since it avoids interacting with the render queue just to query state, but I will defer to Arun as long as he is comfortable with the current approach.